### PR TITLE
PPF-230 endpoint for ppf

### PR DESCRIPTION
**Pull request by JonasVHG**
### Added

-  Added public endpoint `POST /project/{uuid}` to import Project from `publiq-platform`
- `ImportProject` Command  
- `CommandHandler` for `ImportProject`

---
Ticket: https://jira.publiq.be/browse/PPF-230

### CODE REVIEW

File: src/Project/CommandHandler/ImportProjectCommandHandler.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        }
+
+        $this->entityManager->flush();
+
+        $localUser->setFirstName($this->user->givenName);
+        $localUser->setLastName($this->user->familyName);
+        $localUser->setEmail($this->user->mbox);
+        $localUser->setNick($this->user->nick);
+
+        $projectImported = new ProjectImported($project, $localUser);
```
Comment (LucWollants) [inline src/Project/CommandHandler/ImportProjectCommandHandler.php:108]: I wonder if we need to dispatch this event? I don't think we need to do anything extra except create this new widget project inside the database?

Comment (LucWollants) [reply src/Project/CommandHandler/ImportProjectCommandHandler.php]: As discussed, currently no need to dispatch a `ProjectImported` event.

File: src/Project/Controller/ProjectControllerTrait.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace CultuurNet\ProjectAanvraag\Project\Controller;
+
+use CultuurNet\ProjectAanvraag\Core\Exception\MissingRequiredFieldsException;
+
+trait ProjectControllerTrait
```
Comment (LucWollants) [inline src/Project/Controller/ProjectControllerTrait.php:7]: I would calls this `ValidateFields` or something similar.

File: test/Project/Controller/ImportProjectControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+    /**
+     * @var Request & MockObject
+     */
+    private $request;
+
+    /**
+     * @var \stdClass
+     */
+    private $formData;
```
Comment (LucWollants) [inline test/Project/Controller/ImportProjectControllerTest.php:55]: No need for this reusable object, just do it inline and make it more readable.

File: src/Project/Controller/ProjectControllerTrait.php (hunk 2 above)
Comment (JonasVHG) [reply src/Project/Controller/ProjectControllerTrait.php:7]: I orginallly called it `ProjectControllerRequest` but the `ci` demanded that it should have a name ending with `Trait`, so I just used the same name for Clarity. 
ValidateFieldsTrait could offcourse be a good alternative.

File: test/Project/Controller/ImportProjectControllerTest.php [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+use CultuurNet\ProjectAanvraag\Coupon\CouponValidatorInterface;
+use CultuurNet\ProjectAanvraag\Entity\Project;
+use CultuurNet\ProjectAanvraag\Entity\ProjectInterface;
+use CultuurNet\ProjectAanvraag\Insightly\InsightlyClientInterface;
+use CultuurNet\ProjectAanvraag\Insightly\Item\Contact;
+use CultuurNet\ProjectAanvraag\Insightly\Item\Link;
+use CultuurNet\ProjectAanvraag\Insightly\Item\Organisation;
+use CultuurNet\ProjectAanvraag\Integrations\Insightly\InsightlyClient;
+use CultuurNet\ProjectAanvraag\Integrations\Insightly\PipelineStages;
+use CultuurNet\ProjectAanvraag\Project\Command\ActivateProject;
```
Comment (LucWollants) [inline test/Project/Controller/ImportProjectControllerTest.php:16]: I think there are a lot of unused namespaces here.

File: test/Project/Controller/ProjectControllerTest.php [hunk 5]
```diff
@@ -17,6 +17,7 @@
 use CultuurNet\ProjectAanvraag\Project\Command\BlockProject;
 use CultuurNet\ProjectAanvraag\Project\Command\CreateProject;
 use CultuurNet\ProjectAanvraag\Project\Command\DeleteProject;
+use CultuurNet\ProjectAanvraag\Project\Command\ImportProject;
```
Comment (LucWollants) [inline test/Project/Controller/ProjectControllerTest.php:20]: Unused now.

File: test/Project/Controller/ImportProjectControllerTest.php (hunk 4 above)
Comment (JonasVHG) [reply test/Project/Controller/ImportProjectControllerTest.php]: Indeed, I had too much faith in `make ci`

**Review by LucWollants — APPROVED** [review]

**Pull request merged**